codec-haproxy/src/main/java/io/netty/handler/codec/haproxy/HAProxyMessage.java

```diff
@@ -117,16 +98,15 @@ private HAProxyMessage(
      * @return                           {@link HAProxyMessage} instance
      * @throws HAProxyProtocolException  if any portion of the header is invalid
      */
-    static HAProxyMessage decodeHeader(ByteBuf header) {
-        if (header == null) {
-            throw new NullPointerException("header");
-        }
-
+    static HAProxyMessage decodeStructHeader(ByteBuf header) {
+        checkNotNull(header, "header");
         if (header.readableBytes() < 16) {
             throw new HAProxyProtocolException(
                     "incomplete header: " + header.readableBytes() + " bytes (expected: 16+ bytes)");
         }
 
+        ByteBuf rawContent = header.retain();
```

Comment:
imho we should call `retain` as last method to ensure we can not leak

Reply:
Done :-)